Shared/Extensions/URLExtensions.swift

```diff
@@ -468,6 +468,39 @@ extension URL {
     }
 }
 
+extension URL {
+    public var isBookmarklet: Bool {
+        return self.absoluteString.isBookmarklet
+    }
+    
+    public var bookmarkletCodeComponent: String? {
+        return self.absoluteString.bookmarkletCodeComponent
+    }
+}
+
+extension String {
+    public var isBookmarklet: Bool {
+        let url = self.lowercased()
+        return url.hasPrefix("javascript:") && !url.hasPrefix("javascript:/")
```

Comment:
FWIW there are potentially other ways to bypass this (embedded tabs, etc): https://www.owasp.org/index.php/XSS_Filter_Evasion_Cheat_Sheet

if firefox has some kind of internal functionality for detecting bookmarklets / js urls, that would be preferable

however i don't think there is any security impact to a URL not being detected as a bookmarklet? IIUC that just means nothing will happen when it is entered.

Reply:
@diracdeltas I tried quite a few of these. They don't work.

I create a script that injects a tag into each page and attempts some of the above:

```javascript
var img = document.createElement("img");
img.src = "jav  ascript:window.location.href='https://brave.com';";
document.body.appendChild(img);
```

```javascript
var img = document.createElement("img");
img.src = "jav\tascript:window.location.href='https://brave.com';";
document.body.appendChild(img);
```

```javascript
var img = document.createElement("img");
img.src = "jav\nascript:window.location.href='https://brave.com';";
document.body.appendChild(img);
```

```javascript
var img = document.createElement("img");
img.src = "jav&#x09;ascript:window.location.href='https://brave.com';";
document.body.appendChild(img);
```

```javascript
location.href="javascript:window.location.href='https://brave.com'";
```

etc..